### DISCUSSION TITLE
Get it to work with openHAB

### DISCUSSION THREAD

**Discussion by obones** [toverainc/willow#106] [Q&A] [answered]
Hello,

I have just built and flashed willow to my ESP Box and while it recognizes my phrases, it keeps on talking to my openHAB server with URLs that are not expected for it. I mean, here is what I see from willow:

```
I (06:16:56.813) WILLOW: HASS URL: http://1.2.3.4:8080/api/conversation/process
I (06:16:56.821) WILLOW: sending '{"text":"collect strawberries","language":"en"}' to Home Assistant API on 'http://1.2.3.4:8080/api/conversation/process'
I (06:16:56.847) WILLOW: HTTP status='302' content_length='0'
```

Now, in this [thread](https://community.openhab.org/t/willow-open-source-echo-google-home-quality-speech-hardware-for-50/146717/21) over at openHAB community forums, you mention the following curl done by willow:

```
curl -u $OH_TOKEN: https://$OH_INSTANCE/rest/voice/interpreters -H 'accept: application/json' -H 'Accept-Language: en' -H 'Content-Type: text/plain' -d 'turn off upstairs fan'
```

Obviously, I missed something during configuration but it wasn't that obvious to me.
Can you tell me what I should to to have it talk the OH dialect?

**Comment by kristiankielhofner** [comment, marked as answer]
You have Willow configured for Home Assistant but you are trying to use it with openHAB. You need to select openHAB from the "Command Endpoint" section of Willow Configuration and fill in your openHAB server address and API key as described in the README.

**Reply by obones** [reply]
Ah yes, indeed. For reference, here it is:
![image](https://github.com/toverainc/willow/assets/1314739/ca17a824-88d4-4ed1-ad79-2815eb354798)

Note that I missed it because it is indented and my brain assumed that it is related to the inference server and simply ignored it.

**Reply by kristiankielhofner** [reply]
It's indented because that is how the underlying configuration system (Kconfig) handles multiple-choice options such as this. It may be able to orient left but we haven't looked into it.
